python/cudf/cudf/core/reshape.py

```diff
@@ -449,11 +470,16 @@ def length_check(obj, name):
     else:
         result_df = df.drop(labels=columns)
         for name in columns:
-            if hasattr(df[name]._column, "categories"):
+            if isinstance(df[name]._column, CategoricalColumn):
                 unique = df[name]._column.categories
             else:
                 unique = df[name].unique()
 
+            if dummy_na is False:
+                if not is_string_dtype(unique):
+                    unique = unique.nans_to_nulls()
```

Comment:
Shouldn't we only need to do this if `unique` is a `float` type?

Reply:
I was thinking about categorical as well, but as we have already filtered that out, only float should suffice. Nice, catch.